context/videosContext.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+import { IVideoArr, IVideoContext } from "../types";
+
+export const VideoContext = createContext({} as IVideoContext);
+
+export const VideoProvider: React.FC = ({ children }) => {
```

Comment:
Does this have to be in a context? I can only see that it's being used in a single place.

What do you think of removing this context and the use of `useVideos`, and instead move the fetch into `getServerSideProps()` function in `pages/index.tsx` (where `useVideos` is currently used) and use `YOUTUBE_API` instad of `NEXT_PUBLIC_YOUTUBE_API` to keep you key secure (removing prefix `NEXT_PUBLIC_` will make environment variable only usable server side).

This will fetch the youtube videos server side and provide the data to the client when they fetches the page.

Reply:
That's a great suggestion. I believe I'll be needing the `useVideos` hook a lot many times in the project ahead.

But, if it doesn't turn out that way, I'll surely switch to your suggestion.
Thank you.